src/legacy/core_plugins/data/public/search/search_service.ts

```diff
@@ -63,28 +76,41 @@ export interface SearchStart {
  * it will move into the existing search service in src/plugins/data/public/search
  */
 export class SearchService {
+  private readonly aggTypesRegistry = new AggTypesRegistry();
+
   public setup(core: CoreSetup): SearchSetup {
+    const aggTypesSetup = this.aggTypesRegistry.setup();
+    aggTypes.buckets.forEach(b => aggTypesSetup.registerBucket(b));
+    aggTypes.metrics.forEach(m => aggTypesSetup.registerMetric(m));
+
     return {
       aggs: {
-        types: aggTypes, // TODO convert to registry
-        // TODO add other items as needed
+        types: aggTypesSetup,
       },
     };
   }
 
   public start(core: CoreStart): SearchStart {
+    const aggTypesStart = this.aggTypesRegistry.start();
     return {
       aggs: {
-        types: aggTypes, // TODO convert to registry
-        AggConfig, // TODO make static
-        AggConfigs,
-        AggType,
-        aggTypeFieldFilters,
-        FieldParamType,
-        MetricAggType,
-        parentPipelineAggHelper, // TODO make static
-        siblingPipelineAggHelper, // TODO make static
-        setBounds, // TODO make static
+        createAggConfigs: (indexPattern, configStates = [], schemas) => {
+          return new AggConfigs(indexPattern, configStates, {
+            schemas,
+            typesRegistry: aggTypesStart,
+          });
+        },
+        types: aggTypesStart,
+        __LEGACY: {
+          AggConfig, // TODO make static
+          AggType,
+          aggTypeFieldFilters,
+          FieldParamType,
+          MetricAggType,
+          parentPipelineAggHelper, // TODO make static
+          setBounds, // TODO make static
+          siblingPipelineAggHelper, // TODO make static
+        },
```

Comment:
This section will be addressed in the next PR so we can remove the `ui/agg_types` directory